scripts/update

```diff
@@ -1,8 +1,63 @@
-# If current pwd is scripts, go 1 up.
-if [ ${PWD##*/} == "scripts" ]; then
-    cd ..
+#! /bin/bash
+
+# update [HA_SRC_DIR [PID [LOG_FILE [PID_FILE]]]]
+# HA_SRC_DIR - The directory where the Home Assistant source lives
+# PID - The Process ID of Home Assistant if it is currently running. If this
+#       is supplied, Home Assistant will be shutdown before updating and then
+#       restarted after the update has been applied.
+# LOG_FILE - A file where the information usually printed to the screen should
```

Comment:
I think there are two ways to accomplish this.

1) Is to create a Daemon Helper component that will manage PID and Log files. This seems a bit heavy for such a simple feature.
2) Create PID and LOG flags that HA will interpret in __main__ when launching.

I can't think of any other way we could have the locations of the PID and Log files accessible from inside HA, but there may very well be something I'm not considering. I personally would rather the second option.